### PR TITLE
serializability and 2PL

### PR DESCRIPTION
**Pull request by linjonathan**

### CODE REVIEW

File: transactions/README.md [hunk 1]
```diff
@@ -17,7 +17,39 @@ This allows us to implement **Write-Ahead Logging** (WAL):
...
+**Exclusive**: Need to have before writing to an object  
+
+We can't have a protocol that allows independent grabbing of locks, as non-serializable schedules could still be permitted (strawman).
+
+## 2-Phase Locking
+Once a transaction has **released** a lock it is **not allowed to obtain** any other locks. Therefore, there are two phases, a **growing phase** when transactions acquire locks, and a **shrinking phase** when transactions release locks. This ensures serializability!  
+![serial schedule image](/transactions/2pl_example.png)  
+
+### Issues:
+* While 2-PL precludes non-serializable schedules, it also precludes some serializable schedules (not an exact one-to-one mapping).
```
Comment (vlad17) [inline transactions/README.md:48]: i think you need a newline before L47

File: transactions/README.md [hunk 2]
```diff
@@ -17,7 +17,39 @@ This allows us to implement **Write-Ahead Logging** (WAL):
 ### Concurrency Control
 **Schedule** for transactions is an ordering of the operations performed by those transaction. We want to find two equivalent schedules, i.e. they order all **conflicting** operations in the same way.
 
+**Serial Schedule**: a serial execution of two (or more) transactions.  
+![serial schedule image](/transactions/serial_schedule.png)
+
+**Conflict Serializable**: when a schedule is equivalent to **some** serial schedule
+
+But how do we test for serializability? We create a **precedence graph**, where each node **t** represents a transaction **t**, and there is an edge from **s** to **t** if some action **s** precedes and conflicts with some action of **t**. Then, a schedule is **conflict-serializable** if and only if its precedence graph is **acyclic.**
```
Comment (vlad17) [inline transactions/README.md]: if you're refering to a code entity (e.g. s or t), can you use back ticks `s`, `t`? only vocab should be bold.

**Review by vlad17 — CHANGES_REQUESTED** [review]

File: transactions/README.md (hunk 1 above)
Comment (linjonathan) [reply transactions/README.md:48]: works fine but added

**Review by vlad17 — APPROVED** [review]
double back ticks unnecessary

**Pull request merged**